modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -40,69 +44,80 @@ package object engine {
-    case _ => put(Queue.State.init(engine.Queue(List(seq))))
-  }
+  def load(id: Sequence.Id, seq: Sequence[Action]): Handle[Unit] =
+    StateT[Task, EngineState, Unit] { s =>
+      Task {
+        s.get(id).map(t => t.status match {
+          case Status.Running => (s, ())
+          case _ => (s.updated(id, Sequence.State.init(seq)), ())
+        }).getOrElse((s.updated(id, Sequence.State.init(seq)), ()))
+      }
+    }
+
 
   /**
-    * Adds the current Execution` to the completed `Queue`, makes the next
+    * Adds the current `Execution` to the completed `Queue`, makes the next
     * pending `Execution` the current one, and initiates the actual execution.
     *
     * If there are no more pending `Execution`s, it emits the `Finished` event.
     */
-  def next(q: EventQueue): Handle[Unit] =
-    gets(_.next).flatMap {
-      // Empty state
-      case None     => send(q)(finished)
-      // Final State
-      case Some(qs: Queue.State.Final) => put(qs) *> send(q)(finished)
-      // Execution completed, execute next actions
-      case Some(qs) => put(qs) *> execute(q)
-    }
+  def next(q: EventQueue)(id: Sequence.Id): Handle[Unit] = getS(id).flatMap(_.map { seq =>
+      seq.next match {
+        // Empty state
+        case None => send(q)(finished(id))
+        // Final State
+        case Some(qs: Sequence.State.Final) => putS(id)(qs) *> send(q)(finished(id))
+        // Execution completed, execute next actions
+        case Some(qs) => putS(id)(qs) *> execute(q)(id)
+      }
+    }.getOrElse(unit)
+  )
+
 
   /**
     * Checks the `Status` is `Running` and executes all actions in the `Current`
     * `Execution` in parallel. When all are done it emits the `Executed` event.
     * It also updates the `State` as needed.
     */
-  private def execute(q: EventQueue): Handle[Unit] = {
+  private def execute(q: EventQueue)(id: Sequence.Id): Handle[Unit] = {
 
     // Send the expected event when the `Action` is executed
     def act(t: (Action, Int)): Task[Unit] = t match {
       case (action, i) =>
         action.flatMap {
-          case Result.OK(r)    => q.enqueueOne(completed(i, r))
-          case Result.Error(e) => q.enqueueOne(failed(i, e))
+          case Result.OK(r)    => q.enqueueOne(completed(id, i, r))
+          case Result.Error(e) => q.enqueueOne(failed(id, i, e))
         }
     }
 
-    status.flatMap {
-      case Status.Waiting   => unit
-      case Status.Completed => unit
-      case Status.Running   => (
-        gets(_.current.actions).flatMap(
-          actions => Nondeterminism[Task].gatherUnordered(
-            actions.zipWithIndex.map(act)
-          ).liftM[HandleStateT]
-        )
-      ) *> send(q)(executed)
-    }
+    getS(id).flatMap(_.map { seq =>
+        seq.status match {
+          case Status.Waiting => unit
+          case Status.Completed => unit
+          case Status.Running => {
+            val a = Nondeterminism[Task].gatherUnordered(seq.current.actions.zipWithIndex.map(act))
```

Comment:
As discussed in another PR, probably we should be explicit about the `ExecutorService` when doing `gatherUnordered`. We could leave it to another PR though, but let's add a `TODO` reminder